odl/solvers/nonsmooth/proximal_operators.py

```diff
@@ -1142,10 +1154,68 @@ def proximal_l1(space, lam=1, g=None, isotropic=False):
     [BC2011] Bauschke, H H, and Combettes, P L. *Convex analysis and
     monotone operator theory in Hilbert spaces*. Springer, 2011.
     """
-    # TODO: optimize
-    prox_cc_l1 = proximal_convex_conj_l1(space, lam=lam, g=g,
-                                         isotropic=isotropic)
-    return proximal_convex_conj(prox_cc_l1)
+    lam = float(lam)
+
+    if g is not None and g not in space:
+        raise TypeError('{!r} is not an element of {!r}'.format(g, space))
+
+    class ProximalL1(Operator):
+
+        """Proximal operator of the l1-norm/distance."""
+
+        def __init__(self, sigma):
+            """Initialize a new instance.
+
+            Parameters
+            ----------
+            sigma : positive float
+                Step size parameter
+            """
+            super(ProximalL1, self).__init__(
+                domain=space, range=space, linear=False)
+            self.sigma = float(sigma)
+
+        def _call(self, x, out):
+            """Apply the operator to ``x`` and stores the result in ``out``."""
+            # diff = x - g
+            if g is not None:
+                diff = x - g
+            else:
+                if x is out:
+                    # Handle aliased data properly
+                    diff = x.copy()
+                else:
+                    diff = x
+
+            if isotropic:
```

Comment:
The isotropic case should be moved to another function imo, it is somewhat fundamentally different.